apps/twap/src/app/plugins/orderbook.ts

```diff
@@ -0,0 +1,110 @@
+import { FastifyInstance } from 'fastify';
+import fp from 'fastify-plugin';
+
+import { Settlement } from '../orderbook/settlement';
+import { DataSource, Repository } from 'typeorm';
+import { Trade } from '../orderbook/trade';
+import { Order } from '../orderbook/order';
+import { SupportedChainId } from '@cowprotocol/cow-sdk';
+import { getApiBaseUrl } from '../utils/getApiBaseUrl';
+import { ExecutionInfo } from '../types/order';
+
+const DEFAULT_TWAP_EXECUTION_INFO = {
+  executedSellAmount: BigInt(0),
+  executedBuyAmount: BigInt(0),
+  executedFeeAmount: BigInt(0),
+};
+
+function getExecutionInfoFactory(dataSource: DataSource, apiBaseUrl: string) {
+  return async function getExecutionInfo(
+    orderId: string,
+    partIds: string[]
+  ): Promise<ExecutionInfo> {
+    // TODO: We need to create a query here to derive the execution info from the
+    // read only db.
+    // However, because staging db does not have any of the data we need, we can't rely on this.
+    // Based on: https://github.com/cowprotocol/services/blob/main/crates/database/src/orders.rs#L483-L549
+    // Other options are: Subgraph or watch for the settlement event
+    try {
+      const orders: ExecutionInfo[] = await Promise.all(
+        partIds.map((partId) =>
+          fetch(`${apiBaseUrl}/orders/${partId}`)
+            .then((res) => res.json())
+            .then((json) => ({
+              executedBuyAmount: BigInt(json.executedBuyAmount),
+              executedSellAmount: BigInt(json.executedSellAmount),
```

Comment:
You might want to use `executedSellAmountBeforeFees` instead.

Maybe it doesn't matter for non-swap orders, where the fee is taken from the surplus, but still, keep that in mind.

See this TWAP part order example https://api.cow.fi/goerli/api/v1/orders/0x5173a80aa1d97bceb2e0a72ea7ac1de47825c3b157c4c443ae2ec6415607ba385895fcf0a96a4d19e17d2dd759926d1a02781ef864b14a83

Both are the same, but depending on what you querying it won't be.

Reply:
Good point, updating this. cc/ @shoom3301 for visibility